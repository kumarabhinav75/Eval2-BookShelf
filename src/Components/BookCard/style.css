.cardContainer {
  width: 350px;
  margin-top: 30px;
  background: darkorange;
}

.book-image {
  max-width: 100%;
  max-height: 100%;
}

.book-image img {
  width: 100%;
  height: 250px
}


.book-name {
  color: black;
  margin: 10px 50px;
  font-size: 1.1em;
  text-align: center;
  font-weight: bold;
}

.rating-like {
  display: flex;
  justify-content: space-between;
}
.rating-like div {
  height:8%;
  width:8%;
  margin: 10px;
}

.rating  {
  display: flex;
  justify-content: flex-start;
}

.rating span {
  font-weight: bold;
  font-size: 1.2rem;
}

.claps span {
  color: #6d6c6c;
  margin: 0;
  vertical-align: middle;
  font-size: 1.3em;
  margin-left: 5px;
  margin-top: 0;
}