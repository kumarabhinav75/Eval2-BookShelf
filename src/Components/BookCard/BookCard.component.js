import React, {Component} from 'react';
import './style.css';
import redHeart from '../../Icon/heart-red.svg';
import blackHeart from '../../Icon/heart-black.svg';
// const axios = require('axios');

class Card extends Component{
  state = {
    liked: this.props.liked,
  };

  toggleLike = () => {
    return this.setState({
      liked: !(this.state.liked),
    })
  }

  flipLike = () => {
    if(this.state.liked === true) {

      return redHeart;
    }
    else {
      return blackHeart;
    }
  }

  render(){
    console.log('BOOKCARD',this.props.bookData);
      return(
          <div className="cardContainer">
            <section className="book-image">
              <img src={require('../../Image/cover_image.jpeg')} alt="art"/>
            </section>
            <section>
              <p className="book-name">{this.props.bookData.Name}</p>
            </section>
            <section className="book-reaction">
              <div className="rating-like">
                <div className="rating">
                  <span>{this.props.bookData.rating}</span>
                </div>
                <div className="likes">
                  <img src = {this.flipLike()} alt="clap" onClick = {this.toggleLike}/>
                </div>
              </div>
            </section>
          </div>
      );

  }
}

export default Card;