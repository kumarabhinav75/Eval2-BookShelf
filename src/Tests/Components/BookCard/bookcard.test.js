import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import BookCard from '../../../Components/BookCard/BookCard.component';

describe('BookCard', () => {
  it ('Should match the snapshot', () => {
    const book_data = {
      Name: "Abhinav",
      rating: "4.5"
    }
    const tree = renderer.create(<BookCard bookData = {book_data} />);
    expect(tree).toMatchSnapshot();
  })
})