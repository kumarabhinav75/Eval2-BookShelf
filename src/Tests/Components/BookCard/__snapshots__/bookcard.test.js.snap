// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard Should match the snapshot 1`] = `
<div
  className="cardContainer"
>
  <section
    className="book-image"
  >
    <img
      alt="art"
      src="cover_image.jpeg"
    />
  </section>
  <section>
    <p
      className="book-name"
    >
      Abhinav
    </p>
  </section>
  <section
    className="book-reaction"
  >
    <div
      className="rating-like"
    >
      <div
        className="rating"
      >
        <span>
          4.5
        </span>
      </div>
      <div
        className="likes"
      >
        <img
          alt="clap"
          onClick={[Function]}
          src="heart-black.svg"
        />
      </div>
    </div>
  </section>
</div>
`;
